presidio-analyzer/presidio_analyzer/predefined_recognizers/certificate_number_recognizer.py

```diff
@@ -0,0 +1,77 @@
+from typing import List, Tuple, Optional
+
+from presidio_analyzer import Pattern, PatternRecognizer
+
+
+class CertificateNumberRecognizer(PatternRecognizer):
+    """
+    Recognize common Medical license numbers using regex + checksum.
+
+    :param patterns: List of patterns to be used by this recognizer
+    :param context: List of context words to increase confidence in detection
+    :param supported_language: Language this recognizer supports
+    :param supported_entity: The entity this recognizer can detect
+    :param replacement_pairs: List of tuples with potential replacement values
+    for different strings to be used during pattern matching.
+    This can allow a greater variety in input, for example by removing dashes or spaces.
+    """
+
+    PATTERNS = [
+        Pattern(
+            "USA DEA Certificate Number (weak)",
+            r"[a-zA-Z]{2}\d{7} | [a-zA-Z]{1}9\d{7}",   
+            0.3,
+        ),
+    ]
+
+    CONTEXT = [
+        "medical",
+        "certificate",
+        "DEA"
+    ]
+
+    def __init__(
+        self,
+        patterns: Optional[List[Pattern]] = None,
+        context: Optional[List[str]] = None,
+        supported_language: str = "en",
+        supported_entity: str = "Certificate_Number",
+        replacement_pairs: Optional[List[Tuple[str, str]]] = None,
+    ):
+
+        self.replacement_pairs = (
+            replacement_pairs if replacement_pairs else [("-", ""), (" ", "")]
+        )
+        context = context if context else self.CONTEXT
+        patterns = patterns if patterns else self.PATTERNS
+        super().__init__(
+            supported_entity=supported_entity,
+            patterns=patterns,
+            context=context,
+            supported_language=supported_language,
+        )
+
+    def validate_result(self, pattern_text: str) -> bool:  # noqa D102
+        sanitized_value = self.__sanitize_value(pattern_text, self.replacement_pairs)
+        checksum = self.__luhn_checksum(sanitized_value)
```

Comment:
How would this handle the one/two letters at the beginning? `__luhn_checksum` assumes a numeric string if I understand correctly.

Reply:
that was a good point, I modify it to only pass the digit parts to the check sum.